### PR TITLE
Change list endpoint parameters

### PR DESCRIPTION
**Pull request by jenssegers**
Introduce a `filter` and `page` parameter to list endpoints.

### CODE REVIEW

**Comment by jenssegers** [comment]
@jdrieghe Changed!

**Comment by jenssegers** [comment]
Question; what do we do with the `companyTypes.list` endpoint which requires a country parameter. Do we add a `filter` here as well?

**Comment by jdrieghe** [comment]
I think we decided to only have optional parameters in the `filter` array and add everything else as a regular one.

File: apiary.apib [hunk 1]
```diff
@@ -104,17 +104,19 @@ Get a list of all tags
...
 Get a paginated list of contacts
 
 + Parameters
 
-    + amount: `20` (number, optional) - Amount of contacts you want returned, 1 to 100.
-        + Default: `20`
-    + pageno: `0` (number, optional) - The numerated page you want, starts at 0.
-        + Default: `0`
-    + company_id: `1` (number, optional) - Filter contacts by a certain company.
+    + filter (object)
```
Comment (Brammm) [inline apiary.apib]: Don't know if it's possible in API blueprint, but would make these top level objects optional too.

```
+ filter (object, optional)
+ page (object, optional)
```

This comment counts for all the files where this was changed.

File: apiary.apib [hunk 2]
```diff
@@ -281,7 +281,7 @@ Search companies
         + filter (object, optional)
             + tag: `upsell` (string, optional) - Company tag. Teamleader will only return companies that have the tag.
             + modified_since: `2017-04-26T10:01:49+00:00` (string, optional) - ISO 8601 date. Teamleader will only return contacts that have been added or modified since that date.
-            + query: `teamleader` (string, optional) - A search string. Teamleader will try to match each part of the string to the forename, surname, company name and email address.
+            + term: `teamleader` (string, optional) - A search string. Teamleader will try to match each part of the string to the forename, surname, company name and email address.
```
Comment (jdrieghe) [inline apiary.apib]: 👌  I LIKE

**Pull request merged**